src/OrleansRuntime/Messaging/OutboundMessageQueue.cs

```diff
@@ -125,6 +127,22 @@ public void SendMessage(Message msg)
                     return;
                 }
 
+                // check for simulation of lost messages
+                if (Silo.CurrentSilo.TestHookup.SimulatedMessageLoss != null)
```

Comment:
(We should continue discussion in #1128 which is a separate PR for all the testing-related functionality)

This is used in an automated unit test for the global-single-instance grain (it tests that grains are properly marked as doubtful, and then when communication is restored, the protocol retries and succeeds in going to owned state).

BTW, I do not currently have unit tests that use a setting other than 100% loss, but I see no reason to add that sometime.